sql-odbc/src/sqlodbc/opensearch_communication.cpp

```diff
@@ -433,57 +434,71 @@ OpenSearchCommunication::IssueRequest(
+    std::string error_type = error_details->source_type;
+    if (error_type =="SQLFeatureDisabledException") {
+        return false;
+    }
+    return true;
+}
+
+bool OpenSearchCommunication::CheckSQLPluginAvailability() {
+    LogMsg(OPENSEARCH_ALL, "Checking for SQL plugin status.");
+    std::string test_query = "SELECT 1";
     try {
-        rabbit::document doc;
-        doc.parse(plugin_response);
-
-        rabbit::array plugin_array = doc;
-        for (auto it : plugin_array) {
-            if (it.has("component") && it.has("version")) {
-                std::string plugin_name = it.at("component").as_string();
-                if (!plugin_name.compare(OPENSEARCH_SQL_PLUGIN_NAME)) {
-                    std::string sql_plugin_version =
-                        it.at("version").as_string();
-                    LogMsg(OPENSEARCH_INFO, std::string("Found SQL plugin version '"
-                                                + sql_plugin_version + "'.")
-                                        .c_str());
-                    return true;
-                }
-            } else {
-                m_error_message =
-                    "Could not find all necessary fields in the plugin "
-                    "response object. "
-                    "(\"component\", \"version\")";
+        std::shared_ptr< Aws::Http::HttpResponse > response =
+            IssueRequest(SQL_ENDPOINT_FORMAT_RAW,
+                         Aws::Http::HttpMethod::HTTP_POST, ctype, test_query);
+        if (response == nullptr) {
+            m_error_message =
+                "Failed to receive response."
+                "Received NULL response.";
+            SetErrorDetails("Execution error", m_error_message,
+                            ConnErrorType::CONN_ERROR_QUERY_SYNTAX);
+            LogMsg(OPENSEARCH_ERROR, m_error_message.c_str());
+            return false;
+        }
+
+        AwsHttpResponseToString(response, m_response_str);
+        if (response->GetResponseCode() == Aws::Http::HttpResponseCode::OK) {
+            return true;
+        }
+        else {
+            std::unique_ptr< OpenSearchResult > result =
+                std::make_unique< OpenSearchResult >();
+            AwsHttpResponseToString(response, result->result_json);
+            std::shared_ptr< ErrorDetails > error_details =
+                ParseErrorResponse(*result);
+
+            if(!IsSQLPluginEnabled(error_details)) {
+                m_error_message +=
+                    "The SQL plugin is disabled. The SQL plugin must be "
+                    "enabled in order to use this driver. Response body: '"
+                    + m_response_str + "'";
+                SetErrorDetails("Connection error", m_error_message,
+                                ConnErrorType::CONN_ERROR_COMM_LINK_FAILURE);
+            }
+
+            if (response->HasClientError()) {
+                m_error_message += " Client error: '"
```

Comment:
What is the  action item if the customer receive "Client error" or "Response error". Could we collapsed into single error message.